src/PlaneWaveBasis.jl

```diff
@@ -377,8 +377,7 @@ end
     mul!(f_real, basis.ipFFT, f_real)
 
     # Truncate
-    fill!(f_fourier, 0)
-    f_fourier[:] = f_real[kpt.mapping]
+    f_fourier .= view(f_real, kpt.mapping)
```

Comment:
This should be documented, since now if f_fourier contains garbage (is not fully allocated to zero), then the resulting data will be garbage. Did you check we don't use the property that f_fourier can be undefined as input to this function?

Reply:
No, it's OK: this line actually fills the whole of `f_fourier` (f_fourier is a subset of f_real)

Reply:
Ah sorry that's on the kpoint-specific FFT than it's ok (and I wonder how we missed that until now).